### DISCUSSION TITLE
Webstorm is complaing about createEffect function argument.

### DISCUSSION THREAD

**Discussion by AhmedBenyahia** [ngrx/platform#3752] [Q&A]
I have this code in my application that is highlighted in red. I could not resolve it without adding a type to the arrow function.
I am not sure if this is a problem with ngRx or Webstorm. I tried common solutions like deleting node_module.

My Ngrx dependencies 

```
"@ngrx/effects": "13.0.2",
    "@ngrx/store": "13.0.2",
    "@ngrx/store-devtools": "13.0.2"
```
The error displayed

`Argument type () => Observable<TypedAction<string> | ({searchResults: RpmSearchResults} & TypedAction<"[RPM] Search eUICC records success">)> is not assignable to parameter type () => (Observable<TypedAction<string> | ({searchResults: RpmSearchResults} & TypedAction<"[RPM] Search eUICC records success">)> & ConditionallyDisallowActionCreator<DispatchType<EffectConfig>, Observable<TypedAction<string> | ({searchResults: RpmSearchResults} & TypedAction<"[RPM] Search eUICC records success">)>>)   Type Observable<TypedAction<string> | ({searchResults: RpmSearchResults} & TypedAction<"[RPM] Search eUICC records success">)> is not assignable to type Observable<TypedAction<string> | ({searchResults: RpmSearchResults} & TypedAction<"[RPM] Search eUICC records success">)> & ConditionallyDisallowActionCreator<DispatchType<EffectConfig>, Observable<TypedAction<string> | ({searchResults: RpmSearchResults} & TypedAction<"[RPM] Search eUICC records success">)>>     Type Observable<TypedAction<string> | ({searchResults: RpmSearchResults} & TypedAction<"[RPM] Search eUICC records success">)> is not assignable to type Observable<TypedAction<string> | ({searchResults: RpmSearchResults} & TypedAction<"[RPM] Search eUICC records success">)> & "ActionCreator cannot be dispatched. Did you forget to call the action creator function?" `

![image](https://user-images.githubusercontent.com/16466499/213206224-70e8acb2-59dc-40d5-9b1e-6a559f632c0b.png)

**Comment by timdeschryver** [comment]
Do you have a shareable reproduction @AhmedBenyahia ?
As far as I know there is no problem with WebStorm.
Did you also try VSCode?

I suppose that there's something wrong with the code, but I'm not sure.
You have to make sure that all paths return an `Action`, I'm suspicious of `this.showError`.

**Reply by e-oz** [reply]
I can't reproduce either

**Reply by AhmedBenyahia** [reply]
Those are my environment configuration for context:

```
WebStorm 2022.2
Build #WS-222.3345.108, built on July 22, 2022
Licensed to ...
You have a perpetual fallback license for this version.
Subscription is active until September 1, 2023.
Runtime version: 17.0.3+7-b469.32 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Linux 5.15.0-58-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 4048M
Cores: 8
Registry:
    ide.balloon.shadow.size=0

Non-Bundled Plugins:
    com.mallowigi (65.1.0)
    com.chrisrm.idea.MaterialThemeUI (7.7.0)

Current Desktop: ubuntu:GNOME
```

**Reply by AhmedBenyahia** [reply]
This is the code for the showError function. It does return an action. All the effects actually have this error not just this one.
![image](https://user-images.githubusercontent.com/16466499/213997065-def6c8f7-caf6-4a75-a80a-0c2fe331cbe8.png)
